social-circus/src/redux/actions/uploadImageActions.jsx

```diff
@@ -1,27 +1,30 @@
+import { toast } from "react-hot-toast";
 import { getDownloadURL, ref, uploadBytesResumable } from "firebase/storage";
 import { storage } from "../../firebase";
 
-export const UploadImage = async (folder, file) => {
-  const storageRef = ref(storage, `${folder}/${file.name}`);
-  const uploadTask = await uploadBytesResumable(storageRef, file);
-  const path = uploadTask?.ref?.toString();
-  console.log(path);
-  //Please ignore the below comments || WIP
-  //   const link = await DownloadImage(path);
-  //   console.log(link);
-  //   return link;
-
-  //   const imageRef = ref(storage, `posts/${id.uid}/post-cover.jpg`);
-  //   const response = await uploadBytes(imageRef, image);
-  //   const picRef = ref(storage, `posts/${id.uid}/post-cover.jpg`);
-  //   const imageResponse = await getMetadata(picRef);
-  //   setImagePath(imageResponse);
+export const UploadImage = async (path, file) => {
+  try {
+    const imageRef = ref(storage, path);
+    const response = await uploadBytesResumable(imageRef, file);
+    const pathName = response?.ref?.toString();
+    const gsReference = ref(storage, pathName);
+    const url = await getDownloadURL(gsReference);
+    return url;
+  } catch (error) {
+    console.log(error);
```

Comment:
you can add ToastMsg here.

Reply:
Ah. Yes, cause the user wouldn't know if the image got uploaded. Nice Vaibhav.